client/main/instance.js

```diff
@@ -328,7 +328,165 @@ const Instance = Backbone.Model.extend({
             default:
                 return '';
         }
-    }
+    },
+    toPB(options, Messages) {
```

Comment:
hey, these functions (`toPB`, `fromPB`, `_toPB` and `_fromPB`) strike me as good candidates for shifting into a separate file. they have simple interfaces, and we're always trying to slim down these uber classes like `Instance`